validation/scenarios/test_full_loop.py

```
@@ -0,0 +1,93 @@
+    restart_snapshot = validator2.metrics.snapshots[-1]
+
+    # Verify memory survived restart
+    assert_memory_persisted(initial_snapshot, restart_snapshot)
+
+    # Run 10 more calls with evolved prompt
+    result2 = validator2.run_scenario(
+        tasks=CODING_TASKS[30:40],
+        max_calls=10,
+    )
+
+    final_snapshot = validator2.metrics.snapshots[-1]
```

Ruff (0.15.0)

[error] Local variable `final_snapshot` is assigned to but never used

Remove assignment to unused variable `final_snapshot`

(F841)